src/basis/LexingUtil.ml

```diff
@@ -4,11 +4,12 @@ type span =
   {start : Lexing.position;
    stop : Lexing.position}
 
-let pp_span : span Pp.printer =
-  fun fmt span ->
-  Format.fprintf fmt "%a:%i.%i-%i.%i"
+let pp_span : (span * string) Pp.printer =
+  fun fmt (span,last_token) ->
+  Format.fprintf fmt "%a:%i.%i-%i.%i near %s"
```

Comment:
Perhaps the last token should not be printed with the span, but instead should be part of the `ParseError` message etc. Then, the span output protocol can remain the same, but we can report "There was a parsing error near [token]" or something.

Reply:
I've changed it in the latest commit to keep the info line the same and just add error messages near the pp data line.